lib/libzfs/libzfs_sendrecv.c

```diff
@@ -2422,6 +2422,10 @@ zfs_send(zfs_handle_t *zhp, const char *fromsnap, const char *tosnap,
 		}
 		zfs_handle_t *tosnap = zfs_open(zhp->zfs_hdl,
 		    full_tosnap_name, ZFS_TYPE_SNAPSHOT);
+		if (tosnap == NULL) {
+			err = -1;
```

Comment:
err = EZFS_BADPATH;

Reply:
> err = EZFS_BADPATH;

This is not necessarily true, `zfs_open()` can fail for many different reasons, not just invalid input.

Reply:
Thats right. IRC this was how the old code handled it. Would have to look again to be sure though. Nonetheless I'm fine either way.